src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4148,4 +4153,18 @@ int getClassFileVersion() {
      int getClassFileAccessFlags() {
          return classFileAccessFlags;
      }
+
+    // Checks whether the class name exceeds the maximum allowed length.
+    private static boolean classNameLengthIsValid(String name) {
+	    Objects.requireNonNull(name);
+	    return name.length() <= JAVA_CLASSNAME_MAX_LEN;
```

Comment:
This check is incorrect, as the `JAVA_CLASSNAME_MAX_LEN` applies to the Modified UTF‑8 length, rather than the UTF‑16 length.